### PR TITLE
Feature/np navigation component

### PR DESCRIPTION
**Pull request by mnieznaj**
set up vuex state structure and logic for navigation component

### CODE REVIEW

File: src/components/Navigation/FoodCategories/FoodCategories.vue [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+export default defineComponent({
+  props: {
+    categories: {
+      type: Array
+    },
+    activeCat: {
+      type: String,
+      default: "all"
+    //  zrobić default all z ikoną restauracji to będzie pokazywać wszystkie opcje
```
Comment (entrop-) [inline src/components/Navigation/FoodCategories/FoodCategories.vue]: if you type //todo in comment then you can browse all comments like that more easly in your IDE

File: src/components/Navigation/FoodCategories/FoodCategories.vue [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    return {
+      activeCategory
+    };
+  },
+  methods: {
+    altText(name: string): string {
+      return name + " icon";
+    },
+    setActiveCategory(event: MouseEvent) {
+      console.log(event);
```
Comment (entrop-) [inline src/components/Navigation/FoodCategories/FoodCategories.vue]: plz remove

File: src/components/Navigation/FoodCategories/FoodCategories.vue [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  }
+});
+</script>
+
+<style lang="scss" scoped>
+.np-category-list{
+  @apply flex flex-row justify-start w-full;
+
+  height: 60px;
+  margin-bottom: 40px;
```
Comment (entrop-) [inline src/components/Navigation/FoodCategories/FoodCategories.vue:62]: try to find nearest rem value and avoid px values

File: src/components/Navigation/FoodCategories/FoodCategories.vue [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+  &__item {
+    @apply flex flex-row items-center;
+
+    color: theme("colors.secondary.DEFAULT");
```
Comment (entrop-) [inline src/components/Navigation/FoodCategories/FoodCategories.vue:67]: color-secondary doesnt work?

File: src/components/Navigation/SearchBar/SearchBar.vue [hunk 5]
```diff
@@ -0,0 +1,253 @@
+<template>
+  <section class="np-search-bar">
+    <button type="button"
+            class="np-search-bar__toggle-list-view"
+            @click="changeListView">
+      <icon-base icon-name="list" class-name="np-search-bar__list-view-icon">
+        <ListIcon />
+      </icon-base>
+<!--      <svg enable-background="new 0 0 24 24"-->
+<!--           height="100" viewBox="0 0 24 24" width="100" xmlns="http://www.w3.org/2000/svg"><path d="m5 0h-4c-.552 0-1 .448-1 1v4c0 .552.448 1 1 1h4c.552 0 1-.448 1-1v-4c0-.552-.448-1-1-1z"/><path d="m23 0h-14c-.552 0-1 .448-1 1v4c0 .552.448 1 1 1h14c.552 0 1-.448 1-1v-4c0-.552-.448-1-1-1z"/><path d="m5 9h-4c-.552 0-1 .448-1 1v4c0 .552.448 1 1 1h4c.552 0 1-.448 1-1v-4c0-.552-.448-1-1-1z"/><path d="m23 9h-14c-.552 0-1 .448-1 1v4c0 .552.448 1 1 1h14c.552 0 1-.448 1-1v-4c0-.552-.448-1-1-1z"/><path d="m5 18h-4c-.552 0-1 .448-1 1v4c0 .552.448 1 1 1h4c.552 0 1-.448 1-1v-4c0-.552-.448-1-1-1z"/><path d="m23 18h-14c-.552 0-1 .448-1 1v4c0 .552.448 1 1 1h14c.552 0 1-.448 1-1v-4c0-.552-.448-1-1-1z"/></svg>-->
```
Comment (entrop-) [inline src/components/Navigation/SearchBar/SearchBar.vue]: dont commit commented out code, just remove it

File: src/components/Navigation/SearchBar/SearchBar.vue [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+            </g>
+
+          </svg>
+        </button>
+      </div>
+      <ul class="np-search-bar__filter-list"
+          :class="isLiActive ? 'np-search-bar__filter-list--active' : ''"
+          @click="setActiveFilter">
+
+        <li class="np-search-bar__filter-list-item"><span>Price </span>Low to High</li>
```
Comment (entrop-) [inline src/components/Navigation/SearchBar/SearchBar.vue]: use translation when displaying strings of any kind (ask Szymon)

File: src/components/Navigation/SearchBar/SearchBar.vue [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+      </ul>
+    </section>
+
+  </section>
+</template>
+
+<script lang="ts">
+import IconBase from "@/assets/IconBase.vue";
+import ListIcon from "@/assets/icons/list.vue";
+// import SearchIcon from "@/assets/icons/loupe.vue";
```
Comment (entrop-) [inline src/components/Navigation/SearchBar/SearchBar.vue]: remove commented-out code

File: src/store/navigationInterface.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  readonly name: string;
+  readonly categoryType: string;
+  readonly iconUrl: string;
+}
+
+export interface NavStoreModule {
+  state: () => NavState;
+  mutations: {};
+  actions: {};
+  getters: {};
```
Comment (entrop-) [inline src/store/navigationInterface.ts:38]: remove empty objects

File: src/store/navigationModule.ts [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    setFoodList(state: NavState, list: FoodListItem[]) {
+      state.fullFoodList = list;
+      state.filteredFoodList = state.fullFoodList;
+    },
+    orderFoodList(state: NavState) {
+      const foodList = [...state.filteredFoodList];
+      const orderList = () => {
+        switch (state.foodListOrder) {
+          case "alph_asc":
+            foodList.sort((a: any, b: any) => {
```
Comment (entrop-) [inline src/store/navigationModule.ts]: avoid non-descriptive variable names, name them like firstItem, secondItem or whatever it does

File: src/store/navigationModule.ts [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+      const searchedCategory = context.state.activeFoodCategory;
+      const list: FoodListItem[] = await getFoodList(searchedCategory);
+
+      context.commit("setFoodList", list);
+    }
+  },
+  getters: {}
+};
+
+export default navigationModule;
```
Comment (foxy-methoxy) [inline src/store/navigationModule.ts]: a getters żeby dostac sie do produktow?

Comment (mnieznaj) [reply src/store/navigationModule.ts]: w setupie się dostaję za pomocą computed

**Pull request closed without merging**